lib/tictactoe.rb

```diff
@@ -1,9 +1,46 @@
-class TicTacToe 
-  def renderBoard 
-    Array.new(9, '');  
-  end 
+class TicTacToe
+
+  def initialize(io: )
+    @board = Array.new(9, ' ')
+    @currentToken = 'X'
+    @io = io
+  end
+
+  def render_board
+    io.puts "#{@board[0]}|#{@board[1]}|#{@board[2]}"
+    io.puts "------"
+    io.puts "#{@board[3]}|#{@board[4]}|#{@board[5]}"
+    io.puts "------"
+    io.puts "#{@board[6]}|#{@board[7]}|#{@board[8]}"
+    board
+  end
 
   def newGame
-    
-  end 
-end 
+   @board = Array.new(9, ' ')
+  end
+
+ def placeToken(position)
+    board[position] = currentToken
+    render_board
+ end
+
+ def currentPlayer
+    @currentToken
+ end
+
+ def player_turn
```

Comment:
I wonder if there could be a more descriptive name for this method. I tend to prefer to name methods with verbs to help make it clear what action they are performing. (Like how you did with `placeToken`.)

Also, all the methods in this class should probably follow camelCase rather than mixing snake_case with camelCase.